integration/main_test.go

```diff
@@ -0,0 +1,265 @@
+			receivedMessages = append(receivedMessages, d)
+			return rabbitmq.Ack
+		},
+		"test",
+		rabbitmq.WithConsumerOptionsRoutingKey("test"),
+		rabbitmq.WithConsumerOptionsExchangeName(exchange),
+		rabbitmq.WithConsumerOptionsExchangeDeclare,
+	)
+	require.NoError(err)
+	defer consumer.Close()
+
+	pusblisherConn, err := rabbitmq.NewConn(
+		defaultUrl,
+		rabbitmq.WithConnectionOptionsLogging,
+		rabbitmq.WithConnectionOptionsReconnectInterval(reconnectInterval),
+	)
+	require.NoError(err)
+	defer pusblisherConn.Close()
+
+	publisher, err := rabbitmq.NewPublisher(
+		pusblisherConn,
+		rabbitmq.WithPublisherOptionsLogging,
+		rabbitmq.WithPublisherOptionsExchangeName(exchange),
+		rabbitmq.WithPublisherOptionsExchangeDeclare,
+	)
+	require.NoError(err)
+	defer publisher.Close()
+
+	msg := publishTestMessage(require, publisher)
+	publishTestMessage(require, publisher)
+
+	time.Sleep(messageDelay)
+
+	require.Len(receivedMessages, 2)
+
+	var receivedMsg TestMessage
+
+	require.NoError(json.Unmarshal(receivedMessages[0].Body, &receivedMsg))
+	require.Equal(msg, receivedMsg)
+}
+
+func TestBasicReconnect(t *testing.T) {
+	require := require.New(t)
+	require.True(true)
+
+	consumerConn, err := rabbitmq.NewConn(
+		defaultUrl,
+		rabbitmq.WithConnectionOptionsLogging,
+		rabbitmq.WithConnectionOptionsReconnectInterval(reconnectInterval),
+	)
+	require.NoError(err)
+	defer consumerConn.Close()
+
+	receivedMessages := []rabbitmq.Delivery{}
+
+	exchange := fmt.Sprintf("test-%d", rand.Intn(10000))
+
+	consumer, err := rabbitmq.NewConsumer(
+		consumerConn,
+		func(d rabbitmq.Delivery) rabbitmq.Action {
+			receivedMessages = append(receivedMessages, d)
+			return rabbitmq.Ack
+		},
+		"test",
+		rabbitmq.WithConsumerOptionsRoutingKey("test"),
+		rabbitmq.WithConsumerOptionsExchangeName(exchange),
+		rabbitmq.WithConsumerOptionsExchangeDeclare,
+	)
+	require.NoError(err)
+	defer consumer.Close()
+
+	pusblisherConn, err := rabbitmq.NewConn(
+		defaultUrl,
+		rabbitmq.WithConnectionOptionsLogging,
+		rabbitmq.WithConnectionOptionsReconnectInterval(reconnectInterval),
+	)
+	require.NoError(err)
+	defer pusblisherConn.Close()
+
+	publisher, err := rabbitmq.NewPublisher(
+		pusblisherConn,
+		rabbitmq.WithPublisherOptionsLogging,
+		rabbitmq.WithPublisherOptionsExchangeName(exchange),
+		rabbitmq.WithPublisherOptionsExchangeDeclare,
+	)
+	require.NoError(err)
+	defer publisher.Close()
+
+	waitForConnections(require, 2)
+
+	msg := publishTestMessage(require, publisher)
+
+	time.Sleep(messageDelay)
+
+	terminateConnections(require)
+
+	waitForConnections(require, 2)
+
+	publishTestMessage(require, publisher)
+
+	time.Sleep(messageDelay)
+
+	require.Len(receivedMessages, 2)
+
+	var receivedMsg TestMessage
+
+	require.NoError(json.Unmarshal(receivedMessages[0].Body, &receivedMsg))
+	require.Equal(msg, receivedMsg)
+}
+
+func TestBasicReconnectBrokenChannel(t *testing.T) {
+	t.Skip("This test fails currently, but shouldn't")
```

Comment:
Marked this as skipped for now, but from my understand the connection should eventually recover here.